pkg/mounter/oss/ossfs2.go

```diff
@@ -0,0 +1,185 @@
+}
+
+var ossfs2Dbglevels = map[string]string{
+	utils.DebugLevelDebug: "debug",
+	utils.DebugLevelInfo:  "info",
+}
+
+const defaultOssfs2Dbglevel = utils.DebugLevelInfo
+
+func NewFuseOssfs2(configmap *corev1.ConfigMap, m metadata.MetadataProvider) OSSFuseMounterType {
+	config := utils.ExtractFuseContainerConfig(configmap, OssFs2Type)
+	// set default image
+	setDefaultImage(OssFs2Type, m, &config)
+	// set default memory request
+	if _, ok := config.Resources.Requests[corev1.ResourceMemory]; !ok {
+		config.Resources.Requests[corev1.ResourceMemory] = resource.MustParse("50Mi")
+	}
+
+	return &fuseOssfs2{config: config}
+}
+
+func (f *fuseOssfs2) Name() string {
+	return OssFs2Type
+}
+func (f *fuseOssfs2) PrecheckAuthConfig(o *Options, onNode bool) error {
+
+	if o.AuthType != "" {
+		return fmt.Errorf("%s do not support authType: %s", f.Name(), o.AuthType)
+	}
+
+	if features.FunctionalMutableFeatureGate.Enabled(features.RundCSIProtocol3) {
+		return nil
+	}
+	// aksk may retrieve from ENV
+	if onNode && (o.AkID == "" || o.AkSecret == "") {
+		return fmt.Errorf("missing access key in node publish secret")
+	}
+
+	return nil
+}
+
+func (f *fuseOssfs2) MakeAuthConfig(o *Options, m metadata.MetadataProvider) (*utils.AuthConfig, error) {
+	authCfg := &utils.AuthConfig{}
+	authCfg.Secrets = map[string]string{
+		utils.GetPasswdFileName(f.Name()): fmt.Sprintf("--oss_access_key_id=%s\n--oss_access_key_secret=%s", o.AkID, o.AkSecret),
+	}
+
+	return authCfg, nil
+}
+
+func (f *fuseOssfs2) MakeMountOptions(o *Options, m metadata.MetadataProvider) (mountOptions []string, err error) {
+	mountOptions = append(mountOptions, []string{
+		fmt.Sprintf("oss_endpoint=%s", o.URL),
+		fmt.Sprintf("oss_bucket=%s", o.Bucket),
+		fmt.Sprintf("oss_bucket_prefix=%s", o.Path),
+	}...)
+
+	if o.ReadOnly {
+		mountOptions = append(mountOptions, "ro=true")
+	}
+
+	if o.SigVersion == SigV4 {
+		region, _ := m.Get(metadata.RegionID)
+		if region == "" {
+			return nil, fmt.Errorf("SigV4 is not supported without region")
+		}
+		mountOptions = append(mountOptions, fmt.Sprintf("oss_region=%s", region))
+	}
+	return
+}
+
+func (f *fuseOssfs2) PodTemplateSpec(c *utils.FusePodContext, target string) (*corev1.PodTemplateSpec, error) {
+	spec, err := f.buildPodSpec(c, target)
+	if err != nil {
+		return nil, err
+	}
+
+	pod := new(corev1.PodTemplateSpec)
+	pod.Spec = spec
+
+	pod.Annotations = make(map[string]string)
+	for k, v := range f.config.Annotations {
+		pod.Annotations[k] = v
+	}
+	pod.Labels = make(map[string]string)
+	for k, v := range f.config.Labels {
+		pod.Labels[k] = v
+	}
+	return pod, nil
+}
+
+func (f *fuseOssfs2) buildPodSpec(c *utils.FusePodContext, target string) (spec corev1.PodSpec, _ error) {
+	targetDir := filepath.Dir(target)
+	targetDirVolume := corev1.Volume{
+		Name: "target-dir",
+		VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{
+				Path: targetDir,
+				Type: new(corev1.HostPathType),
```

Comment:
```suggestion
				Type: ptr.To(corev1.HostPathDirectoryOrCreate),
```
See https://pkg.go.dev/k8s.io/utils/ptr#To